modules/cnf-programming-the-sriov-operator.adoc

```diff
@@ -0,0 +1,398 @@
+metadata:
+    name: config
+    namespace: vran-acceleration-operators
+    spec:
+      nodes:
+        - nodeName: node1 <1>
+          physicalFunctions:
+            - pciAddress: 0000.1d.00.0 <2>
+              pfDriver: pci-pf-stub
+              vfDriver: vfio-pci
+              vfAmount: 2 <3>
+              bbDevConfig:
+                n3000:
+                  # Network Type: either "FPGA_5GNR" or "FPGA_LTE"
+                  networkType: "FPGA_5GNR"
+                  pfMode: false
+                  flrTimeout: 610
+                  downlink:
+                    bandwidth: 3
+                    loadBalance: 128
+                    queues: <4>
+                      vf0: 16
+                      vf1: 16
+                      vf2: 0
+                      vf3: 0
+                      vf4: 0
+                      vf5: 0
+                      vf6: 0
+                      vf7: 0
+                  uplink:
+                    bandwidth: 3
+                    loadBalance: 128
+                    queues: <5>
+                      vf0: 16
+                      vf1: 16
+                      vf2: 0
+                      vf3: 0
+                      vf4: 0
+                      vf5: 0
+                      vf6: 0
+                      vf7: 0
+----
+<1> Specify the node name.
+<2> Specify the PCI Address of the card on which the SR-IOV-FEC Operator will be installed.
+<3> Specify the number of virtual functions. Create two virtual functions.
+<4> On `vf0` create one queue with 16 buses (downlink and uplink).
+<5> On `vf1` create one queue with 16 buses (downlink and uplink).
++
+[NOTE]
+====
+For Intel® PAC N3000 for vRAN Acceleration the user can create up to 8 VF devices. Each FEC PF device provides a total of 64 queues to be configured, 32 queues for uplink and 32 queues for downlink. The queues would be typically distributed evenly across the VFs.
+====
+.. Apply the CR.
++
+[source,terminal]
+----
+$ oc apply -f sriovfec_n3000.yaml
+----
++
+After creation of the CR, the SR-IOV FEC daemon starts configuring the FEC device.
+
+.. Check the status:
++
+[source,terminal]
+----
+oc get sriovfecclusterconfig config -o yaml
+----
++
+.Example output
+[source,yaml]
+----
+status:
+    conditions:
+    - lastTransitionTime: "2020-12-15T17:19:37Z"
+      message: Configured successfully
+      observedGeneration: 1
+      reason: ConfigurationSucceeded
+      status: "True"
+      type: Configured
+    inventory:
+      sriovAccelerators:
+      - deviceID: 0d8f
+        driver: pci-pf-stub
+        maxVirtualFunctions: 8
+        pciAddress: 0000:1d:00.0
+        vendorID: "8086"
+        virtualFunctions:
+        - deviceID: 0d90
+          driver: vfio-pci
+          pciAddress: 0000:1d:00.1
+        - deviceID: 0d90
+          driver: vfio-pci
+          pciAddress: 0000:1d:00.2
+----
+
+. Check the logs:
+.. Determine the name of the SR-IOV daemon pod:
++
+[source,terminal]
+----
+$ oc get pod | grep sriov-fec-daemonset
```

Comment:
If an SME approves selecting the pods in the `sriov-fec-daemonset` by label, please make the same update here.